src/system.rs

```diff
@@ -122,6 +165,33 @@ fn uptime_internal(data: &str) -> isize {
     FromStr::from_str(uptime[0]).unwrap()
 }
 
+/// Returns the system load average
+///
+/// `/proc/loadavg` contains the system load average
+pub fn loadavg() -> Result<LoadAverage> {
+    let data = read_file(Path::new("/proc/loadavg"))?;
+    // strips off any trailing new line as well
+    let fields: Vec<&str> = data.split_whitespace().collect();
```

Comment:
Could you split out a separate function from this point on, with the signature `fn loadavg_internal(proc_file_contents: &str) -> Result<LoadAverage>`?

That makes this much easier to test - you can call `loadavg_internal` and pass it some sample data and assert specific things about the results, rather than it parsing the system's /proc/uptime and you only being able to assert that it's greater than zero.